client/proposalmsgs.go

```diff
@@ -187,9 +187,9 @@ func (p BaseChannelProposal) NumPeers() int {
 
 // Encode encodes the BaseChannelProposal into an io.Writer.
 func (p BaseChannelProposal) Encode(w io.Writer) error {
-	return perunio.Encode(w,
-		p.ChallengeDuration, p.NonceShare,
-		OptAppAndDataEnc{p.App, p.InitData}, p.InitBals, p.FundingAgreement)
+	optAppAndDataEnc := channel.OptAppAndDataEnc{App: p.App, Data: p.InitData}
```

Comment:
Why not use same struct for encoding and decoding?

Reply:
See https://github.com/hyperledger-labs/go-perun/pull/272#discussion_r763695117